core/build.gradle

```diff
@@ -194,3 +196,43 @@ if (tasks.findByName('trimShadedJar')) {
         dontnote
     }
 }
+
+class PublicSuffixesTask extends DefaultTask {
+    // Rebuild up to once in 7 days, unless cleaned.
+    @Input
+    final def days = (System.currentTimeMillis() / TimeUnit.DAYS.toMillis(7)).intValue()
+    @OutputFile
+    final def publicSuffixesFile = project.file("${project.buildDir}/resources/main/com/linecorp/armeria/public_suffixes.txt")
+
+    @TaskAction
+    def run() {
+        try {
+            def tempFile = File.createTempFile("public_suffixes", null)
+            println "tempFile is created: ${tempFile}"
+
+            def url = 'https://publicsuffix.org/list/public_suffix_list.dat'
+            def req = new URL(url).openConnection()
+            if (req.getResponseCode() != 200) {
+                throw new Exception('Received a non-200 response from publicsuffix.org: ' +
+                        "${req.getResponseCode()}, ${req.getHeaderFields()}")
```

Comment:
How about just warning and skipping when this happens, so that the build continues even if `publicsuffix.org` is down? It should be fine as long as we pushed the previously fetched one into the repository, as I suggested in the above comment.